src/auth/auth.service.ts

```diff
@@ -5,24 +5,28 @@ import {
 } from '@nestjs/common';
 import { UserService } from '../user/user.service';
 import { JwtService } from '@nestjs/jwt';
-import { User } from '@prisma/client';
 import { bcryptConstants } from './constants';
 import * as bcrypt from 'bcrypt';
 import { SignInRequest, SignUpRequest, UserRequest } from './dto';
+import { UserRepository } from '../user/user.repository';
+import { UserPayloadRequest } from '../user/dto';
 
 @Injectable()
 export class AuthService {
   constructor(
     private readonly userService: UserService,
+    private readonly userRepository: UserRepository,
     private readonly jwtService: JwtService,
   ) {}
 
   signIn(signInRequest: SignInRequest): Promise<string> {
     return this.jwtService.signAsync(signInRequest);
   }
 
-  async signUp(signUpRequest: SignUpRequest): Promise<User> {
-    const user = await this.userService.findOneUser(signUpRequest.email);
+  async signUp(signUpRequest: SignUpRequest): Promise<UserPayloadRequest> {
+    const user = await this.userRepository.getUserByEmailWithPassword(
+      signUpRequest.email,
+    );
```

Comment:
here I think we should return `User` as before

Reply:
Actually, when I look at this I'm more confused then before. This at the end calls userService.createUser() which calls userRepository which calls prisma so then Prisma returns User, next userRepository maps it and return same thing, as userService, which passes this thing back to authService (which is UserPayloadRequest). And that's right to me because auth.service should return User without password so controller can send it out.
So should we really change it?

Reply:
ok, so here ideally we should return an object that has only the fields that we need. I don't want to return `UserPayloadRequest` here, because as name suggest it's not a response, so it would be confusing.